blob/s3blob/s3blob.go

```diff
@@ -88,8 +88,7 @@ func init() {
 
 // Set holds Wire providers for this package.
 var Set = wire.NewSet(
-	Options{},
-	URLOpener{},
+	wire.Struct(new(URLOpener), "ConfigProvider"),
```

Comment:
My understanding of having these `Set`s types provides convenient ways for the user to compose their provider sets. I think it is ok to have some defined required fields to use them, however, I'm not sure what the criteria we would use. Is it that the field is required for the type to work at all, or maybe 80% of the time the user would want to use the "sets" of fields we defined?

I think most of the changes here look OK since a lot of them only have one field and it often makes sense to provide it. Maybe the solution here is document it in our design.md.

Reply:
Right. I don't see it as something beta-blocking (we can just add more sets over time), and I'd rather we get usage information before proceeding with it full-scale. My priority was to keep existing sets working in a reasonable fashion.